src/Search/Annotation/GearmanTaskDriver.php

```diff
@@ -150,6 +154,8 @@ public function work()
 
                 return;
             }
+            // Dispatch any outstanding signals.
+            Signals::tick();
```

Comment:
To keep a small interface, this work could be done in `Signals::isValid()` (no matter how it is renamed to uniform interfaces)